stdlib/public/core/String.swift

```diff
@@ -694,10 +694,37 @@ extension String {
   ///   `withCString(_:)` method. The pointer argument is valid only for the
   ///   duration of the method's execution.
   /// - Returns: The return value, if any, of the `body` closure parameter.
+  @abi(
+    func withCString<Result>(
+      _ body: (UnsafePointer<Int8>) throws -> Result
+    ) rethrows -> Result
+  )
   @inlinable // fast-path: already C-string compatible
-  public func withCString<Result>(
+  @usableFromInline
+  internal func __rethrows_withCString<Result>(
     _ body: (UnsafePointer<Int8>) throws -> Result
-  ) rethrows -> Result {
+  ) throws -> Result {
+    return try unsafe _guts.withCString(body)
+  }
+
+  /// Calls the given closure with a pointer to the contents of the string,
+  /// represented as a null-terminated sequence of UTF-8 code units.
+  ///
+  /// The pointer passed as an argument to `body` is valid only during the
+  /// execution of `withCString(_:)`. Do not store or return the pointer for
+  /// later use.
+  ///
+  /// - Parameter body: A closure with a pointer parameter that points to a
+  ///   null-terminated sequence of UTF-8 code units. If `body` has a return
+  ///   value, that value is also used as the return value for the
+  ///   `withCString(_:)` method. The pointer argument is valid only for the
+  ///   duration of the method's execution.
+  /// - Returns: The return value, if any, of the `body` closure parameter.
+  @inlinable // fast-path: already C-string compatible
+  @_alwaysEmitIntoClient
```

Comment:
Small nit: if possible, please avoid listing both `@_alwaysEmitIntoClient` and `@inlinable` -- our usual pattern is to just use `@_aeic` in these cases. (The presence of `@inlinable` usually implies that an ABI symbol gets exported; it gets slightly misleading we start listing it to functions where that's not the case.)